### PR TITLE
add username in RedisLockStore

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Works on https://rasahq.atlassian.net/browse/ATO-1520
- Add `username` to the connection arguments of `RedisLockStore`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by Urkem — APPROVED** [review]
LGTM

File: rasa/core/lock_store.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def __init__(
         host: Text = "localhost",
         port: int = 6379,
         db: int = 1,
+        username: Optional[Text] = None,
```
Comment (ancalita) [inline rasa/core/lock_store.py:203]: How is this property extracted from yaml? Double-checking we don't need to do something else there?

**Review by ancalita — COMMENTED** [review]
Can we please add unit testing and if possible integration testing for the connection with a username?

File: rasa/core/lock_store.py (hunk 1 above)
Comment (vcidst) [reply rasa/core/lock_store.py:203]: I'm guessing that I'll have to add a line for `username` in endpoints.yml and it should work. However I'll do a quick test with docker compose and a testbot.

Once that works out I plan to also add unit tests. I didn't think about integration tests yet but if it doesn't take long -- I'll be sure to add that too.

Comment (ancalita) [reply rasa/core/lock_store.py:203]: > I'll have to add a line for username in endpoints.yml

Yep, exactly 👍🏻

Comment (vcidst) [reply rasa/core/lock_store.py:203]: @ancalita added an integration test. It doesn't look like unit test will be possible with our given testing setup

**Comment by vcidst** [comment]
Closing this PR in favor of https://github.com/RasaHQ/rasa/pull/12835

**Pull request closed without merging**